search-sentiments-map/test/trends-test.js

```diff
@@ -0,0 +1,243 @@
+      assert.deepEqual(results, mockResult1half);
+    });
+
+    it('should retrieve the most recent datastore entry given beyond the given time range', 
+        async function() {
+      let results = await trends.retrieveGlobalTrendsForTimeRange(2);
+      assert.deepEqual(results, mockResult3);
+    });
+  });
+
+  describe('UpdateDailyTrends', function() {
+    let mockDailyTrendsData;
+    let mockTrendsByCountry;
+    let datastoreEntities;
+    const countryJson = require('./../public/countries-with-trends.json');
+
+    beforeEach(() => {
+      datastoreEntities = [];
+
+      let mockTrendingSearches = [
+        {
+          'title': {
+            'query': 'topic1',
+            'exploreLink': 'exploreLink1',
+          },
+          'formattedTraffic': '200K+',
+          'articles': [
+            {
+              'title': 'article1',
+              'url': 'url1',
+            }, {
+              'title': 'article2',
+              'url': 'url2',
+            }
+          ],
+        }
+      ];
+      mockDailyTrendsData = JSON.stringify({
+        'default': {
+          'trendingSearchesDays': [
+            {
+              'trendingSearches': mockTrendingSearches,
+            }
+          ],
+        },
+      });
+
+      mockTrendsByCountry = [];
+      for (let i = 0; i < countryJson.length; i++) {
+        mockTrendsByCountry.push({
+          country: countryJson[i].id,
+          trends: [
+            {
+              topic: 'topic1',
+              traffic: '200K+',
+              exploreLink: 'exploreLink1',
+              articles: [
+                {
+                  title: 'article1',
+                  url: 'url1',
+                }, {
+                  title: 'article2',
+                  url: 'url2',
+                }
+              ],
+            }
+          ],
+        });
+      }
+
+      // Stub the API call to googleTrends.
+      sinon.stub(googleTrends, 'dailyTrends').resolves(mockDailyTrendsData);
+
+      // Stub the `saveTrendsAndDeletePrevious` function.
+      sinon.stub(trends, 'saveTrendsAndDeletePrevious')
+          .callsFake((trendsByCountry) => {
+        datastoreEntities.push({
+          key: 'fakeKey',
+          data: {
+            timestamp: Date.now(),
+            trendsByCountry: trendsByCountry,
+            globalTrends: trends.getGlobalTrends(trendsByCountry),
+          },
+        });
+      });
+    });
+
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should get daily trends and save in datastore', async function() {
+      await trends.updateDailyTrends();
+
+      assert.equal(datastoreEntities[0].key, 'fakeKey');
+      assert.deepEqual(datastoreEntities[0].data.globalTrends, 
+          [{trendTopic: 'topic1', count: countryJson.length}]);
+      assert.deepEqual(
+          datastoreEntities[0].data.trendsByCountry, mockTrendsByCountry);
+    });
+  });
+
+  describe('ConstructCountryTrendsJson', function() {
```

Comment:
This test can be deleted. This function simply formats a json. It is also actually tested in the updateDailyTrends function above, since it is called within that function.

Reply:
I think it's good to test it again separately since those two tests have different emphases

Reply:
But that's not really how this works. If you want to test it separately, then you should stub this function in the updateDailyTrends test. But this function seems sort of strange to test since all it is testing is formatting.

Reply:
But it has two forEach loops so seems complicated enough to test. On the other hand, stubbing needs more work than simply calling it, so I didn't stub it. But I think I can stub it

Reply:
Yeah because right now what you are doing is effectively testing it twice. So either stub it, or don't do this separate test.